src/pages/payment-methods/index.js

```diff
@@ -82,14 +84,79 @@ const InnerDiv = styled.div`
     overflow-x: auto;
     overflow-y: visible;
 `
+
+const getClientResidence = () => {
+    const domain = getDomain()
+    const client_information = getClientInformation(domain)
+    return client_information ? client_information.residence : ''
+}
+
+const getClientCountry = () => {
+    let current_client_country = ''
+    if (isLoggedIn()) {
+        current_client_country = getClientResidence()
+    } else {
+        const [website_status] = useWebsiteStatus()
+        current_client_country = website_status?.clients_country || ''
+    }
+    return current_client_country.toLowerCase()
+}
+const getPaymentsBasedOnCountry = (current_client_country) => {
+    const payments_data_array = payment_data.map((category) => {
+        const filteredData = category.data.filter((item) => {
+            if (item.countries.included.length) {
+                const includedCountries = item.countries.included.map((i) => i.toLowerCase())
+                if (includedCountries.includes(current_client_country)) {
+                    return true
+                } else if (includedCountries.includes('eu')) {
+                    return isEuCountry(current_client_country)
+                }
+                return false
+            } else if (item.countries.excluded.length) {
+                const excludedCountries = item.countries.excluded.map((i) => i.toLowerCase())
+                if (excludedCountries.includes(current_client_country)) {
+                    return false
+                }
+                if (excludedCountries.includes('eu')) {
+                    return !isEuCountry(current_client_country)
+                }
+                return true
+            }
+            return true
+        })
```

Comment:
Can we simplify the if-else block here?

Reply:
Moved the logic into two separate functions.
```
const filterCountry = (countriesArray, client_country) => {
    const countries = countriesArray.map((i) => i.toLowerCase())
    if (countries.includes(client_country)) {
        return true
    } else if (countries.includes('eu')) {
        return isEuCountry(client_country)
    }
    return false
}

const getPaymentsBasedOnCountry = () => {
    const client_country = getClientCountry()
    const payments_data_array = payment_data.map((category) => {
        const filteredData = category.data.filter((item) => {
            if (item.countries.included.length)
                return filterCountry(item.countries.included, client_country)
            else if (item.countries.excluded.length)
                return !filterCountry(item.countries.excluded, client_country)
            return true
        })
        return {
            ...category,
            data: filteredData,
        }
    })
    return payments_data_array.filter((item) => item.data.length)
}
```